src/test/java/emissary/test/core/junit5/ExtractionTest.java

```diff
@@ -173,8 +173,26 @@ protected void checkAnswers(Element el, IBaseDataObject payload, @Nullable List<
             throws DataConversionException {
 
         int numAtt = JDOMUtil.getChildIntValue(el, "numAttachments");
+        long numAttElements = el.getChildren().stream().filter(c -> c.getName().startsWith("att")).count();
+        // check attachments answer file count against payload count
         if (numAtt > -1) {
-            assertEquals(numAtt, attachments != null ? attachments.size() : 0, "Number of attachments in " + tname);
+            assertEquals(numAtt, attachments != null ? attachments.size() : 0,
+                    String.format("Expected <numAttachments> in %s not equal to number of att in payload.", tname));
+        } else if (numAtt == -1 && numAttElements > 0) {
+            assertEquals(numAttElements, attachments != null ? attachments.size() : 0,
+                    String.format("Expected <att#> in %s not equal to number of att in payload.", tname));
+        } else {
+            if (attachments != null && !attachments.isEmpty()) {
+                fail(String.format("%d attachments in payload with no count in answer xml, add matching <numAttachments> count for %s",
+                        attachments.size(), tname));
+            }
+        }
+        // log if <numAttachments> and <att#> don't match in answer file
+        if (numAtt > -1 || numAttElements > 0) {
+            if (Math.toIntExact(numAttElements) != numAtt) {
+                logger.warn("Expected and actual attachment counts in {} not equal. <numAttachments>: {} <att#>: {}", tname,
+                        Math.max(numAtt, 0), numAttElements);
+            }
```

Comment:
I would just remove these log messages I think. Based on the behavior above we expect this scenario and accept it.

Reply:
You're right, I'll remove them